### PR TITLE
fix: correct dependency injection of custom context fields implementing partial __eq__/__ne__

### PR DESCRIPTION
**Pull request by antoinehumbert**
This PR fixes issue #1806

It includes 2 regression tests related to both corrections in `faststream.utils.context.types` for the 2 use cases where value is compared to EMPTY.

Note that the issue description is incorrect as the custom field class must implement `__ne__` and not only `__eq__` to trigger the problem. This is the reason for the `User` implementation provided in regression test.

Please, let me know if PR needs some improvements.

### CODE REVIEW

**Review by Lancetnik — DISMISSED** [review]

**Comment by Lancetnik** [comment]
LGTM, thank you!

**Comment by Lancetnik** [comment]
> Sorry for not executing the pre-commit check before submitting PR. Do you want me to correct those errors ?

It will be really kind

File: examples/e12_custom_context.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+from typing import TYPE_CHECKING, Any
```
Comment (Lancetnik) [inline examples/e12_custom_context.py:1]: Can you move your example to just a `tests/utils/context/test_main.py` 2 tests? It is too complex to be an example

**Review by Lancetnik — CHANGES_REQUESTED** [review]

**Review by Lancetnik — APPROVED** [review]

**Comment by Lancetnik** [comment]
@antoinehumbert can you run precommit please? 😄

**Comment by antoinehumbert** [comment]
Will do tomorrow. Sorry for the incontinence.

**Comment by antoinehumbert** [comment]
Hi! Everything looks good now.

**Review by Lancetnik — APPROVED** [review]

**Pull request merged**